tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -1104,6 +1104,30 @@ def test_log_softmax_invalid(device, a_shape, axis, dtype):
         return chainerx.log_softmax(a, axis=axis)
 
 
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'shape': [(), (0,), (1,), (2, 0, 3), (1, 1, 1), (2, 3)],
+        'expected_dtypes': _expected_float_dtypes_math_functions,
+        'input': [-2, 0, 2],
+        'contiguous': [None, 'C'],
+    })
+    # Special values
+    + chainer.testing.product({
+        'shape': [(2, 3)],
+        'expected_dtypes': _expected_float_dtypes_math_functions,
```

Comment:
Ditto.

Reply:
Thanks for the review @hvy . Have made the required changes.

Reply:
Thanks for the quick fix! I noticed this latter product is testing infs and nans. Maybe it can be limited to floats as you first wrote. Sorry about the bad comment.

Reply:
Sure! I can revert to the old version. Could I know if there is any specific reason for removing the inf and nan tests?

Reply:
Ah, I meant to test `_expected_float_dtypes_math_functions` for `'input': [-2, 0, 2]` but `numeric_dtypes` for `input': [float('inf'), -float('inf'), float('nan')]`.

Reply:
Using `UnaryMathTestBase` is completely fine. Sorry but what do you mean by removing the inf and nan tests?

Reply:
I completely misunderstood your comment on limiting the inf and nan tests to floats. Will make the changes. Sorry for the inconvenience.